src/SearchIndex.hs

```diff
@@ -277,6 +277,8 @@ extractChildDeclarationType declTitle declInfo cdeclInfo =
             , P.constraintArgs = map (P.TypeVar () . fst) args
             , P.constraintData = Nothing
             , P.constraintAnn = ()
+              -- TODO: Verify that defaulting to `kindType` works
+            , P.constraintKindArgs = map (fromMaybe (P.kindType $> ()) . snd) args
```

Comment:
From @hdgarrood wrt testing this out:

> The places I'd be worried about displaying incorrect information are classes where kinds aren't explicitly declared but will be inferred as something other than Type (e.g. Functor), or where kinds are specified in a kind signature but not in a class declaration

For example:

```purs
class X :: (Type -> Type) -> Constraint
class X a
```

Reply:
I've had a chance to test this out and it seems to work just fine -- though we aren't displaying the kind signatures in the documentation.